runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -10768,7 +10768,26 @@ static TR::Register *inlineIntrinsicIndexOf_P10(TR::Node *node, TR::CodeGenerato
       generateTrg1Src2Instruction(cg, TR::InstOpCode::add, node, endPos, endPos, endPos);
       }
 
-   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi, node, arrAddress, array, TR::Compiler->om.contiguousArrayHeaderSizeInBytes());
+   /*
+    * Determine the address of the first byte to read either by loading from dataAddr or adding the header size.
+    * This is followed by adding in the offset.
+    */
+#if defined(J9VM_GC_SPARSE_HEAP_ALLOCATION)
+   if (TR::Compiler->om.isOffHeapAllocationEnabled())
+      {
+      generateTrg1MemInstruction(
+         cg, TR::InstOpCode::ld, node, arrAddress,
+         TR::MemoryReference::createWithDisplacement(
+            cg, array, TR::Compiler->om.offsetOfContiguousDataAddrField(), 8
+         )
+      );
```

Comment:
everything looks good, except weird-looking indentations around here and below. is it better to put the first `)` after `8` and aligning the second `)` with `cg/TR`? you have the precedence of that already below with `addi`.